lib/hammer_cli/abstract.rb

```diff
@@ -341,27 +341,27 @@ def exception_handler_class
     end
 
     def self.desc(desc=nil)
-      @desc = desc if desc
-      @desc
+      return @desc ||= nil unless desc
+
+      @desc = desc
     end
 
     def self.command_name(name=nil)
-      if @names && name
-        @names << name if !@names.include?(name)
-      else
-        @names = [name] if name
-      end
-      @names || (superclass.respond_to?(:command_names) ? superclass.command_names : nil)
+      command_names(name)
     end
 
     def self.command_names(*names)
-      @names = names unless names.empty?
-      @names || (superclass.respond_to?(:command_names) ? superclass.command_names : nil)
+      names = names.compact
+      return @names ||= (superclass.respond_to?(:command_names) ? superclass.command_names : nil) if names.empty?
+
+      @names ||= []
+      @names |= names
```

Comment:
This changes the implementation and now caches the response of the superclass. Is that intentional? I'd keep the change minimal:
```suggestion
      @names ||= nil
      @names = names unless names.empty?
      @names || (superclass.respond_to?(:command_names) ? superclass.command_names : nil)